### ISSUE TITLE
screen reader: create FocusScopes for Latenight QML

### ISSUE THREAD

**Issue by ywwg** [mixxxdj/mixxx#17053]
see https://github.com/mixxxdj/mixxx/issues/10789#issuecomment-3678213583

> QT QML has so called [FocusScope](https://doc.qt.io/qt-6/qml-qtquick-focusscope.html). So the areas identified in the previous bullet points should become FocusScopes, active focus should get remembered per each FocusScope. Left and right arrow keys should be used to navigate over the buttons inside the active FocusScope. Complex widgets requiring arrow keys to change its values such as sliders and comboboxes should not be reachable this way. Corresponding dropdown buttons and keyboard controls should be added to move focus to them. Up and down arrow keys should allow ergonomic keyboard navigation between similar FocusScopes. Currently the decks are arranged like this: Deck 1 is top left, deck 2 is top right, deck 3 is bottom left and deck 4 is bottom right. The navigation should work top to bottom left to right relative to the active FocusScope. For example when cue button of the deck 1 is in focus, pressing down arrow key would move the focus tothe cue button from the deck 2.

also see: https://www.vcreatelogic.com/index.php/2023/07/14/what-exactly-does-focusscope-do-in-qml/

> What a FocusScope actually does is, it forwards any focus it receives to that one child item within it whose focus property was hard-coded to true.
> 
> FocusScope is like a proxy. Veteran QtWidgets users would have used [QLabel::setBuddy()](https://doc.qt.io/qt-6/qlabel.html#setBuddy) to forward focus from a QLabel to any QWidget next to it. That’s kind of what FocusScope does. I would have actually preferred the item to be called FocusProxy (or even FocusBuddy) because that’s exactly what it does. As a QML developer, you can package a complex component inside a FocusScope, in which one of the child items has its focus property set to true. The item whose focus property is set to true will receive focus, when the root FocusProxy receives keyboard focus.

This may be a large tracking bug that also needs sub-issues, probably to start we should pick some particular area of the UI and add focus scopes to it.

**Comment by jrdevadattan** [comment]
@ywwg I can help with the LateNight QML FocusScope work. I will map the deck and control sections that need independent focus memory, then prototype `FocusScope` boundaries so left/right navigation moves within a section while up/down moves between matching deck sections in the intended top-to-bottom, left-to-right order. I will also avoid putting sliders and combo boxes directly into arrow-key traversal, since those controls need arrow keys for value changes.
